pkg/forwarder/internal/retry/on_disk_retry_queue.go

```diff
@@ -95,8 +95,10 @@ func (s *onDiskRetryQueue) Store(transactions []transaction.Transaction) error {
 		_ = os.Remove(file.Name())
 		return err
 	}
-	defer file.Close()
-
+	err = file.Close()
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	err = file.Close()
	if err != nil {
	    _ = os.Remove(file.Name())
		return err
	}
```
I guess we also have to delete the file in case of `Close` error.

Reply:
It would make sense to clean up the `tmp` file but I wonder if this change should be introduced in this PR. I see similar logic in other parts of the code without removing the files.